### PR TITLE
Adapt burrowbeat for usage

### PR DESCRIPTION
**Pull request by ghost**
Implemented:
- Adapted burrowbeat to libbeat 5.5.1
- Dynamic consumer groups
- Changed format for consumer groups better support in kibana (from nested to one doc per partition/topic)

### CODE REVIEW

File: _meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- key: burrowbeat
+  title: burrowbeat
+  description:
+  fields:
+    - name: counter
```
Comment (mdelaney) [inline _meta/fields.yml]: Looks like this file needs to be updated with the fields this beat sends

File: _meta/kibana/index-pattern/burrowbeat.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  "fieldFormatMap": "{\"@timestamp\": {\"id\": \"date\"}}", 
+  "timeFieldName": "@timestamp", 
+  "title": "burrowbeat-*"
+}
```
Comment (mdelaney) [inline _meta/kibana/index-pattern/burrowbeat.json]: we need a newline at the end of file.

File: burrowbeat.template-es6x.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      }
+    }
+  },
+  "order": 0,
+  "settings": {
+    "index.mapping.total_fields.limit": 10000,
+    "index.refresh_interval": "5s"
+  },
+  "template": "burrowbeat-*"
+}
```
Comment (mdelaney) [inline burrowbeat.template-es6x.json]: we need a newline at the end of file.

File: burrowbeat.template.json [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      }
+    }
+  },
+  "order": 0,
+  "settings": {
+    "index.mapping.total_fields.limit": 10000,
+    "index.refresh_interval": "5s"
+  },
+  "template": "burrowbeat-*"
+}
```
Comment (mdelaney) [inline burrowbeat.template.json]: we need a newline at the end of file.

File: beater/burrowbeat.go [hunk 5]
```diff
@@ -11,15 +11,15 @@ import (
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
-	"github.com/elastic/beats/libbeat/publisher"
+	pub "github.com/elastic/beats/libbeat/publisher"
```
Comment (mdelaney) [inline beater/burrowbeat.go]: why change this line and line 22? This gets us absolutely nothing from what I see.

File: beater/burrowbeat.go [hunk 6]
```diff
@@ -11,15 +11,15 @@ import (
...
 
-	"github.com/goomzee/burrowbeat/config"
+	"github.com/Goomzee/burrowbeat/config"
```
Comment (mdelaney) [inline beater/burrowbeat.go]: AFAIK this is not case sensitive, why change this? (this comment applies to all the places you changed this)

File: beater/burrowbeat.go [hunk 7]
```diff
@@ -57,7 +57,28 @@ func (bt *Burrowbeat) Run(b *beat.Beat) error {
 		case <-ticker.C:
 		}
 
-		for _, group := range bt.groups {
+		logp.Debug("main", "Running tick")
+		groups := bt.groups
+		if len(groups) == 0 {
+			endpoint := "http://" + bt.host + ":" + bt.port + "/v2/kafka/" + bt.cluster + "/consumer"
```
Comment (mdelaney) [inline beater/burrowbeat.go]: Most of the endpoint is the same as below. Maybe create a variable like `endpoint_base` and then add the suffix as required when you use it.

File: beater/burrowbeat.go [hunk 8]
```diff
@@ -57,7 +57,28 @@ func (bt *Burrowbeat) Run(b *beat.Beat) error {
...
+                        resp, err := http.Get(endpoint)
```
Comment (mdelaney) [inline beater/burrowbeat.go]: You are mixing spaces and tabs in the indenting. Go standardizes on tabs, please convert this. You should also run `gofmt` to ensure that things are formatted correctly.

File: beater/burrowbeat.go [hunk 9]
```diff
@@ -86,21 +107,29 @@ func (bt *Burrowbeat) Stop() {
...
-		"@timestamp":		common.Time(time.Now()),
-		"type":			"consumer_group",
-		"count":		1,
-		"cluster":		bt.cluster,
-		"group":		group,
-		"total_partitions":	total_partitions,
-		"total_lag":		total_lag,
-		"burrow_status":	status,
+	partitions := status["partitions"].([]interface{})
+	for i, _ := range partitions {
```
Comment (mdelaney) [inline beater/burrowbeat.go]: Why use the index and discard the value here? The very next line you index into `partitions` with the index, just do `for _, partition := partitions {`

File: burrowbeat.template-es2x.json [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        }
+      }
+    }
+  },
+  "order": 0,
+  "settings": {
+    "index.refresh_interval": "5s"
+  },
+  "template": "burrowbeat-*"
+}
```
Comment (mdelaney) [inline burrowbeat.template-es2x.json]: we need a newline at the end of file.

**Review by mdelaney — CHANGES_REQUESTED** [review]

File: beater/burrowbeat.go (hunk 6 above)
Comment (ghost) [reply beater/burrowbeat.go]: Fixed, was temporary fix for build package

File: beater/burrowbeat.go (hunk 5 above)
Comment (ghost) [reply beater/burrowbeat.go]: for now nothing, but if we move to libbeat 6x there will be need to rename

**Comment by ghost** [comment]
@mdelaney updated according to your comments, could you please look again?

**Comment by ghost** [comment]
sorry missed your comment for endpoint, added endpoint_base and fixed possible crash due when request for groups will fail

File: beater/burrowbeat.go (hunk 5 above)
Comment (mdelaney) [reply beater/burrowbeat.go]: Lets remove this change. If we migrate this to libbeat 6.x there are more changes that will be necessary. For example, ES 6 deprecates the use of types. This PR is about moving this beat to libbeat 5.x.

**Review by mdelaney — APPROVED** [review]
Once you address the last remaining commit it lgtm.

**Pull request merged**